src/data-access/facades/repository-facade.ts

```diff
@@ -250,22 +250,27 @@ export class RepositoryFacade {
     return new MongoSystemInformationRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()), new MongoEntityConverter(LoggerFacade.createLogger()))
   }
 
-  public static createDeviceDataChangedListener(): DataChangedListener<Device> {
+  public static createDeviceDataChangedListener(): DataChangedListener<CoreDevice> {
     return new MongoDeviceChangedListener(
       MongoDatabase.getInstance(LoggerFacade.createLogger()),
       new MongoEntityConverter(LoggerFacade.createLogger()),
       LoggerFacade.createLogger()
     )
   }
 
-  public static createDeviceRepository(): DeviceRepository {
-    return new MongoDeviceRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()), new MongoEntityConverter(LoggerFacade.createLogger()))
+  public static createCoreDeviceRepository(): CoreDeviceRepository {
+    return new MongoCoreDeviceRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()), new MongoEntityConverter(LoggerFacade.createLogger()))
   }
 
   public static createStatusMessageRepository(): StatusMessageRepository {
     return new MongoStatusMessageRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()))
   }
 
+  public static createDeviceRepository(): MongoDeviceRepository {
+    return new MongoDeviceRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()))
+    //throw new Error('Method not implemented.')
+  }
```

Comment:
The type of the mehtod should be the abstraction `DeviceRepository`, not the concrete implementation as you have right now.

Also, remove old, commented out, code :)